### PR TITLE
Feat/mobility

### PR DESCRIPTION
**Pull request by sourabhsaini11**
Personal-loan flow 3, 4 & 5

### CODE REVIEW

File: utils/FIS/FIS12/fisChecks.ts [hunk 1]
```diff
@@ -256,12 +257,12 @@ export const validateContext = (context: any, msgIdSet: any, pastCall: any, cure
       if (curentCall.startsWith('on_')) {
         logger.info(`Comparing Message Ids of /${pastCall} and /${curentCall}`)
         if (!_.isEqual(prevContext.message_id, context.message_id)) {
-          errorObj.message_id = `Message Id for /${pastCall} and /${curentCall} api should be same`
+          errorObj.message_id = `message_id for /${pastCall} and /${curentCall} api should be same`
```
Comment (bluecypher) [inline utils/FIS/FIS12/fisChecks.ts:260]: Why message_id is same for two different calls ?

File: utils/FIS/FIS12/fisChecks.ts [hunk 2]
```diff
@@ -289,3 +290,146 @@ export const validateContext = (context: any, msgIdSet: any, pastCall: any, cure
...
+      errorObj.paymentsAmountMismatch = `Total payments amount (${totalPaymentsAmount}) does not match with quote.price.value (${quotePriceValue})`
+    }
+
+    for (let i = 1; i < payments.length; i++) {
+      const payment = payments[i]
+
+      if (!payment.status) {
+        errorObj.payments = `status is missing in payments`
+      } else {
+        const allowedStatusValues = ['NOT_PAID', 'PAID', 'NOT-PAID']
```
Comment (bluecypher) [inline utils/FIS/FIS12/fisChecks.ts:374]: why are both NOT-PAID and NOT_PAID enums allowed?

File: utils/FIS/FIS12/fisChecks.ts [hunk 3]
```diff
@@ -289,3 +290,146 @@ export const validateContext = (context: any, msgIdSet: any, pastCall: any, cure
...
+
+    if (!payments[0].collected_by) {
+      errorObj.payments = `collected_by  is missing in payments`
+    } else {
+      const allowedCollectedByValues = ['BPP', 'BAP']
+      const allowedStatusValues = ['NOT_PAID', 'PAID', 'NOT-PAID']
+
+      const collectedBy = getValue(`collected_by`)
+      if (collectedBy && collectedBy !== payments[0].collected_by) {
+        errorObj.collectedBy = `Collected_By didn't matched with what was send in previous call.`
```
Comment (bluecypher) [inline utils/FIS/FIS12/fisChecks.ts:417]: pls check grammatical errors: didn't match*.....what was sent*

File: utils/FIS/FIS12/onStatus.ts [hunk 4]
```diff
@@ -236,7 +234,7 @@ export const checkOnStatus = (data: any, msgIdSet: any, flow: string) => {
         onStatusObj.payments = `collected_by  is missing in payments`
       } else {
         const allowedCollectedByValues = ['BPP', 'BAP']
-        const allowedStatusValues = ['NOT_PAID', 'PAID']
+        const allowedStatusValues = ['NOT_PAID', 'PAID', 'NOT-PAID']
```
Comment (bluecypher) [inline utils/FIS/FIS12/onStatus.ts:237]: either one of NOT-PAID/NOT_PAID should be allowed

File: utils/mobility/status.ts [hunk 5]
```diff
@@ -29,11 +29,11 @@ export const checkStatus = (data: any, msgIdSet: any) => {
     Object.assign(errorObj, contextRes.ERRORS)
   }
 
-  if (!message.ref_id) {
-    const key = `${mobilitySequence.STATUS}_ref_id`
-    errorObj[key] = `ref_id in /${constants.STATUS} must be present`
+  if (!message?.ref_id && !message?.order_id) {
+    const key = `${mobilitySequence.STATUS}_id`
+    errorObj[key] = `either of ref_id or order_id must be present in /${constants.STATUS}`
```
Comment (bluecypher) [inline utils/mobility/status.ts:34]: is this allowed?

**Review by bluecypher — CHANGES_REQUESTED** [review]
Please check the feedback and resubmit the PR

@sourabhsaini11

File: utils/FIS/FIS12/fisChecks.ts (hunk 1 above)
Comment (sourabhsaini11) [reply utils/FIS/FIS12/fisChecks.ts:260]: This was to validate if the message_id of a req call & it's async res call is same

File: utils/mobility/status.ts (hunk 5 above)
Comment (sourabhsaini11) [reply utils/mobility/status.ts:34]: If a buyer wants to fetch the latest status for a transactional journey he can do so by sending either of ref_id(order not created) or order_id(order created)

File: utils/FIS/FIS12/fisChecks.ts (hunk 3 above)
Comment (sourabhsaini11) [reply utils/FIS/FIS12/fisChecks.ts:417]: Fixed in latest commit

**Pull request closed without merging**